src/scene/sprite-nine-slice/NiceSliceSpritePipe.ts

```diff
@@ -0,0 +1,122 @@
+import { ExtensionType } from '../../extensions/Extensions';
+import { BigPool } from '../../utils/pool/PoolGroup';
+import { BatchableMesh } from '../mesh/shared/BatchableMesh';
+import { NineSliceGeometry } from './NineSliceGeometry';
+
+import type { InstructionSet } from '../../rendering/renderers/shared/instructions/InstructionSet';
+import type { RenderPipe } from '../../rendering/renderers/shared/instructions/RenderPipe';
+import type { Renderer } from '../../rendering/renderers/types';
+import type { PoolItem } from '../../utils/pool/Pool';
+import type { NineSliceSprite } from './NineSliceSprite';
+
+export class NineSliceSpritePipe implements RenderPipe<NineSliceSprite>
+{
+    /** @ignore */
+    public static extension = {
+        type: [
+            ExtensionType.WebGLPipes,
+            ExtensionType.WebGPUPipes,
+            ExtensionType.CanvasPipes,
+        ],
+        name: 'nineSliceSprite',
+    } as const;
+
+    private readonly _renderer: Renderer;
+    private readonly _gpuSpriteHash: Record<number, BatchableMesh> = Object.create(null);
+
+    constructor(renderer: Renderer)
+    {
+        this._renderer = renderer;
+    }
+
+    public addRenderable(sprite: NineSliceSprite, _instructionSet: InstructionSet)
+    {
+        const gpuSprite = this._getGpuSprite(sprite);
```

Comment:
some of these functions have exactly the same code as `SpritePipe`, should we extract these out into reusable functions?

Reply:
to be followed up in a separate pr